internal/controller/progressive/progressive.go

```diff
@@ -85,6 +90,15 @@ func ProcessResource(
 
 	numaLogger := logger.FromContext(ctx)
 
+	// Make sure that our Promoted Child Status reflects the current promoted child
+	promotedChildStatus := rolloutObject.GetPromotedChildStatus()
+	if promotedChildStatus == nil || promotedChildStatus.Name != existingPromotedChild.GetName() {
+		err := rolloutObject.ResetPromotedChildStatus(existingPromotedChild)
+		if err != nil {
+			return false, false, 0, err
+		}
+	}
```

Comment:
I think we can just call this once at the beginning if either the PromotedChildStatus is nil or if there's a new promoted child. 

By calling `ResetPromotedChildStatus()`, we can update the [InterstepBufferServiceName](https://github.com/numaproj/numaplane/blob/pipeline-ref-to-isbsvc/pkg/apis/numaplane/v1alpha1/pipelinerollout_types.go#L69) for UpgradingPipelineStatus

Reply:
With this, we basically reset the Promoted status whenever there's a new promoted child that starts the upgrading process.